ptychonn/pospred/core.py

```diff
@@ -0,0 +1,517 @@
+                this_a = v - self.last_v
+                self.delta_a_bar = this_a - self.last_a
+                self.last_a = this_a
+            elif self.i_iter == 1:
+                this_a = v - self.last_v
+                self.last_a = this_a
+            else:
+                self.delta_a_bar = np.array([0., 0.])
+        if self.order == 2:
+            if self.i_iter > 1:
+                this_a = v - self.last_v
+                self.a_bar = self.beta * self.a_bar + (1 - self.beta) * this_a
+            elif self.i_iter == 1:
+                self.a_bar = v - self.last_v
+            else:
+                self.a_bar = np.array([0., 0.])
+        if self.order == 1:
+            if self.i_iter > 0:
+                self.v_bar = self.beta * self.v_bar + (1 - self.beta) * v
+            else:
+                self.v_bar = v
+        self.last_v = v
+        self.update_step_length_bar(v)
+        self.i_iter += 1
+
+    def update_offset_with_estimated_value(self, v):
+        """
+        If the offset of the current iteration is estimated, then only update last-iteration values and the
+        iteration number, but do not change any accumulated values.
+
+        :param v: np.ndarray.
+        """
+        if self.order == 3:
+            self.last_a = v - self.last_v
+        self.last_v = v
+        self.i_iter += 1
+
+    def update_step_length_bar(self, v):
+        if self.i_iter == 0:
+            self.step_length_bar = np.linalg.norm(v)
+        else:
+            self.step_length_bar = (self.beta_step_length * self.step_length_bar +
+                                    (1 - self.beta_step_length) * np.linalg.norm(v))
+
+    def estimate(self):
+        this_v = 0.0
+        if self.order == 0:
+            this_v = self.last_v
+        elif self.order == 1:
+            this_v = self.v_bar
+        elif self.order == 2:
+            this_v = self.last_v + self.a_bar
+        elif self.order == 3:
+            this_a = self.last_a + self.delta_a_bar
+            this_v = self.last_v + this_a
+        this_v = this_v / np.linalg.norm(this_v) * self.step_length_bar
+        return this_v
+
+
+class PtychoNNProbePositionCorrector:
+
+    def __init__(self, config_dict: InferenceConfigDict):
+        self.config_dict = config_dict
+        if self.config_dict.reconstruction_image_path is None:
+            if self.config_dict.ptycho_reconstructor is None:
+                self.ptycho_reconstructor = PyTorchReconstructor(self.config_dict)
+            else:
+                self.ptycho_reconstructor = self.config_dict.ptycho_reconstructor
+        else:
+            self.ptycho_reconstructor = VirtualReconstructor(InferenceConfigDict())
+
+        self.dp_data_fhdl = None
+        self.orig_probe_positions = None
+        self.new_probe_positions = None
+        self.n_dps = 0
+        self.debug = self.config_dict.debug
+        self.registrator = None
+        self.method = self.config_dict.method
```

Comment:
```suggestion
        self.registrator = None
```
Just reference `self.config_dict.debug` directly in this class? Why add extra attributes?